lib/main.js

```diff
@@ -70,7 +70,23 @@ globals.setImmediate(() => {
 	runner.on('test', test);
 
 	process.on('ava-run', options => {
-		runner.run(options).then(exit);
+		runner.run(options)
+			.then(exit)
+			.catch(err => {
+				let exception;
+				try {
+					exception = serializeError(err);
+				} catch (_) {
+					// Avoid using serializeError
+					const err = new Error('Runner failed with an unserializable exception');
```

Comment:
It's only called in three places. I'd prefer letting the error propagate correctly, and then treating it as an uncaught exception.

That said I've changed this code to `process.emit('uncaughtException', err)` and moved the fallback stuff to the `uncaughtException` handler in `test-worker.js`.